src/components/LibraryItem.tsx

```diff
@@ -73,31 +90,33 @@ export class LibraryItem extends React.Component<LibraryItemProps, LibraryItemSt
         let nestedElements = null;
         let clusteredElements = null;
 
-        if (this.state.expanded) // Show only nested elements when expanded.
-        {
-            if (this.props.data.childItems && (this.props.data.childItems.length > 0)) {
+        // visible only nested elements when expanded.
+        if (this.state.expanded && this.state.visible && this.props.data.childItems.length > 0) {
```

Comment:
`this.props.data.childItems` condition is removed, is it always non-undefined?

Also this may look cleaner:

```js
let shouldDisplay = this.state.expanded && this.state.visible;
if (shouldDisplay && (this.props.data.childItems.length > 0)) {
    // ...
}
```